huaweicloud/services/as/data_source_huaweicloud_as_configurations.go

```diff
@@ -0,0 +1,231 @@
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"ecs_group_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"disk": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"size": {
+													Type:     schema.TypeInt,
+													Computed: true,
+												},
+												"volume_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												"disk_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												"kms_id": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"personality": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"path": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+												"content": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+									"public_ip": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"eip": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"ip_type": {
+																Type:     schema.TypeString,
+																Computed: true,
+															},
+															"bandwidth": {
+																Type:     schema.TypeList,
+																Computed: true,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"size": {
+																			Type:     schema.TypeInt,
+																			Computed: true,
+																		},
+																		"share_type": {
+																			Type:     schema.TypeString,
+																			Computed: true,
+																		},
+																		"charging_mode": {
+																			Type:     schema.TypeString,
+																			Computed: true,
+																		},
+																	},
+																},
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+									"user_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"metadata": {
+										Type:     schema.TypeMap,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
```

Comment:
here you can extract the contents of Elem into a subfunction, which makes it look simpler.
Such as the resource of huaweicloud_as_bandwidth_policy

Reply:
OK